src/robusta/cli/auth.py

```diff
@@ -0,0 +1,156 @@
+import base64
+import json
+import subprocess
+import traceback
+import uuid
+from typing import Optional
+import click_spinner
+from dpath.util import get
+import requests
+import typer
+import yaml
+from cryptography.hazmat.primitives.serialization import Encoding, PrivateFormat, PublicFormat, NoEncryption
+from cryptography.hazmat.primitives.asymmetric import rsa
+from pydantic import BaseModel
+
+from .backend_profile import backend_profile
+from .playbooks_cmd import NAMESPACE_EXPLANATION, get_playbooks_config
+from .utils import namespace_to_kubectl
+
+AUTH_SECRET_NAME = "robusta-auth-config-secret"
+app = typer.Typer()
+
+
+class RSAKeyPair(BaseModel):
+    prv: str
+    pub: str
+
+
+def gen_rsa_pair() -> RSAKeyPair:
+    # generate private/public key pair
+    key = rsa.generate_private_key(public_exponent=65537, key_size=2048)
+
+    # get public key in OpenSSH format
+    public_key = key.public_key().public_bytes(Encoding.PEM, PublicFormat.SubjectPublicKeyInfo)
+
+    # get private key in PEM container format
+    pem = key.private_bytes(
+        encoding=Encoding.PEM,
+        format=PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=NoEncryption()
+    )
+
+    return RSAKeyPair(
+        pub=public_key.decode('utf-8'),
+        prv=pem.decode('utf-8')
+    )
+
+
+def get_auth_config(namespace: str) -> Optional[RSAKeyPair]:
```

Comment:
I'm nitpicking, but I think `get_existing_auth_config` or `get_auth_config_from_cluster` would be better names. They make it clear where the auth is coming from.

Reply:
fixed